lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -289,6 +289,23 @@ ::mlir::LogicalResult mlir::tt::ttnn::ConcatOp::verify() {
 // ReshapeOp
 //===----------------------------------------------------------------------===//
 
+// ReshapeOp verification
+::mlir::LogicalResult mlir::tt::ttnn::RepeatOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getResult().getType();
+
+  // Check that the shape size matches the rank of the output tensor
```

Comment:
can you add more check here? For example we need to check if 
`for all dims d in input | input[d] * shape[d] == output[d]`

Also check that shape has same rank as input or output (either is fine since you already check if input.rank != output.rank)

Reply:
Added proposed checks to verification.